mule-extensions-api/src/main/java/org/mule/runtime/extension/api/notification/NotificationInfo.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.notification;
+
+import static org.mule.runtime.api.notification.AbstractServerNotification.TYPE_INFO;
+
+/**
+ * Holds all notification related information that will be used to fire a desired notification.
+ *
+ * @param <T> the type of data provided
+ * @since 1.1
+ */
+public interface NotificationInfo<T> {
+
+  /**
+   * Defines the type of notification to fire. Options are:
+   * <p><ul>
+   * <li>{@link org.mule.runtime.api.notification.AbstractServerNotification#TYPE_INFO} (default)
+   * <li>{@link org.mule.runtime.api.notification.AbstractServerNotification#TYPE_TRACE}
+   * <li>{@link org.mule.runtime.api.notification.AbstractServerNotification#TYPE_WARNING}
+   * <li>{@link org.mule.runtime.api.notification.AbstractServerNotification#TYPE_ERROR}
+   * <li>{@link org.mule.runtime.api.notification.AbstractServerNotification#TYPE_FATAL}
+   * </ul><p>
+   *
+   * @return the notification type
+   */
+  default String getType() {
+    return TYPE_INFO;
```

Comment:
not Ext-APi friendly. If the extensions really need to know about this, then this should be an enum